SPECS/babeltrace2/00-fix-lttng-live-array-access.patch

```diff
@@ -0,0 +1,23 @@
+diff -Naur babeltrace2-2.0.4/src/plugins/ctf/common/msg-iter/msg-iter.c babeltrace2-2.0.4-copy/src/plugins/ctf/common/msg-iter/msg-iter.c
+--- babeltrace2-2.0.4/src/plugins/ctf/common/msg-iter/msg-iter.c	2022-02-09 16:04:31.452911700 -0800
++++ babeltrace2-2.0.4-copy/src/plugins/ctf/common/msg-iter/msg-iter.c	2022-02-09 16:24:36.138072700 -0800
+@@ -2001,6 +2001,9 @@
+ 	}
+ 
+ 	if (G_UNLIKELY(int_fc->storing_index >= 0)) {
++		if ((uint64_t) int_fc->storing_index >= msg_it->stored_values->len) {
```

Comment:
@mfrw Is this patch created by you or take from the source repo? If you've created on your own can we try to upstream it before committing here?

Reply:
> @mfrw Is this patch created by you or take from the source repo? If you've created on your own can we try to upstream it before committing here?

No, I did not create the patch.
I think you mean @nicogbg instead of me.

Reply:
@0xba1a the patch was made by @nickbopp for v2.01 of babeltrace2. I believe he did raise the issue with the upstream maintainers, but the fix was not added in v2.0.4. (The current version in Mariner 2.0) This PR is meant to re-introduce the patch.